### PR TITLE
Teach signalfx proto to filter bad spans

### PR DESCRIPTION
**Pull request by mdubbyap**
The downstream sink will get them in the context and can do with that what it wants.

 - zipkinv2 with annotations
 - bad traceid
 - bad spanid

### CODE REVIEW

File: protocol/signalfx/trace_zipkin_test.go [hunk 1]
```diff
@@ -1927,7 +1935,11 @@ func TestParseJaegerFromRequest(t *testing.T) {
 		req := httptest.NewRequest("POST", "/v1/trace", bytes.NewReader(payload))
 
 		got, err := ParseJaegerSpansFromRequest(req)
-		assert.Nil(t, err)
+		assert.True(t, spanfilter.IsInvalid(err))
```
Comment (jgheewala) [inline protocol/signalfx/trace_zipkin_test.go:1938]: use So() with Convery

File: protocol/signalfx/trace_zipkin_test.go [hunk 2]
```diff
@@ -1927,7 +1935,11 @@ func TestParseJaegerFromRequest(t *testing.T) {
...
+		sm := err.(*spanfilter.Map)
+		fmt.Println(sm.Error())
```
Comment (jgheewala) [inline protocol/signalfx/trace_zipkin_test.go:1940]: Do you need this fmt ?

File: protocol/signalfx/trace_zipkin_test.go [hunk 3]
```diff
@@ -14,6 +14,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/signalfx/golib/v3/sfxclient/spanfilter"
```
Comment (jgheewala) [inline protocol/signalfx/trace_zipkin_test.go:17]: we really need a way to make this import work better

**Review by jgheewala — APPROVED** [review]
minor inline comments but rest lgtm

File: protocol/signalfx/trace_zipkin.go [hunk 4]
```diff
@@ -78,69 +77,72 @@ func (is *InputSpan) fromZipkinV2() (*trace.Span, error) {
...
-	return &is.Span, nil
+	return &is.Span
 }
 
-func (is *InputSpan) addParentChildSpanReferenceToJaegerSpan(span *jaegerpb.Span) error {
+func (is *InputSpan) addParentChildSpanReferenceToJaegerSpan(span *jaegerpb.Span) {
 	// only add parent/child reference if the parent id can be parsed
-	parentID, err := jaegerpb.SpanIDFromString(*is.Span.ParentID)
-	if err == nil {
+	if parentID, err := jaegerpb.SpanIDFromString(*is.Span.ParentID); err == nil {
```
Comment (jgheewala) [inline protocol/signalfx/trace_zipkin.go:85]: what to do about the error if parentId cannot be parsed

Comment (mdubbyap) [reply protocol/signalfx/trace_zipkin.go:85]: the field isn't required or a violation, if it's garbage, we get rid of it.

**Pull request merged**